### PR TITLE
Add Saved Posts to the new Reader

### PR DESCRIPTION
**Pull request by Gio2018**
#### Also adds the 'Search' on the top right

Ref #13580

To test:
##### Saved posts
- go to the Reader
- Save/unsave posts from any stream other than saved
- verify that the 'Saved' tab shows posts accordingly
- Verify that tapping one 'view all' on the toasts switches to the 'Saved' tab
- In the 'Saved' tab, tap on the 'save' icon, and verify that the undo cell appears and behaves as expected

##### Search
- tap on the 'search' button in the top right corner and verify that the search works same as the old reader.

PR submission checklist:

- [ ] ~~I have considered adding unit tests where possible.~~ will be added in a future PR
- [ ] ~~I have considered adding accessibility improvements for my changes.~~ will be handled in a future PR
- [ ] ~~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~~ Not released

### CODE REVIEW

**Comment by bjtitus** [comment]
I noticed that the header from the previous section appears to be sticking around when switching to Saved, at least when I have nothing saved. This applies to Discover and Following:

<a href="https://user-images.githubusercontent.com/3250/79294154-bd088a80-7e92-11ea-8671-73d82071ccc1.gif"></a>

**Comment by Gio2018** [comment]
> I noticed that the header from the previous section appears to be sticking around when switching to Saved, at least when I have nothing saved. This applies to Discover and Following:
> 

Thanks for catching this!! Totally overlooked to reset that header, should be fixed now.

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewModel.swift [hunk 1]
```diff
@@ -91,6 +88,7 @@ extension ReaderTabViewModel {
 
         self.tabSelectionCallback = { [weak controller] topic in
             controller?.setTopic(topic)
+            controller?.isSavedPostsController = topic == nil ? true : false
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewModel.swift:91]: I don't think the tertiary is necessary. Couldn't it just be this?

`controller?.isSavedPostsController = (topic == nil)`

File: WordPress/Classes/ViewRelated/Reader/ReaderPostCellActions.swift [hunk 2]
```diff
@@ -171,3 +199,24 @@ enum ReaderActionsVisibility: Equatable {
...
+
+    func restoreUnsavedPost(_ post: ReaderPost) {
+        removedPosts.remove(post)
+    }
+
+    func clearRemovedPosts() {
+        let allRemovedPosts = removedPosts.all()
+        for post in allRemovedPosts {
+            toggleSavedForLater(for: post)
+        }
```
[the remark this answers is not in the document]
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/ReaderPostCellActions.swift]: Totally! 
Fixed

File: WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift [hunk 3]
```diff
@@ -6,7 +6,7 @@ enum ReaderSaveForLaterOrigin {
     case postDetail
     case readerMenu
 
-    fileprivate var saveActionValue: String {
+    var saveActionValue: String {
```
[the remark this answers is not in the document]
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift]: True; in reality I only need `viewAllPostsValue` to be visible in `ReaderStreamViewController+Helper`, until the new reader is officially released, so I fixed the other two and put a todo for this

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 4]
```diff
@@ -140,6 +140,23 @@ import WordPressFlux
...
+            }
+        }
+        didSet {
+            if isSavedPostsController {
+                configureControllerForTopic(synchronize: false)
+                trackSavedListAccessed()
+                postCellActions?.visibleConfirmation = false
+            } else {
+                postCellActions?.visibleConfirmation = true
+            }
```
[the remark this answers is not in the document]
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: yes it could!

**Comment by Gio2018** [comment]
> Hey @Gio2018 . This looks nice! Just a few nitpicky comments. Otherwise :shipit: !

Thanks for all the valuable suggestions @ScoutHarris !!

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewModel.swift (hunk 1 above)
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewModel.swift]: 👍

**Pull request merged**